[feature] Custom scan_each_shard batch size

donaldong opened the pull request:
This allows users to decide how many keys to fetch for each scan.

According to redis, the SCAN command
> The default COUNT value is 10

This is not ideal for our use case -- we gonna end up with too many round trips. I updated the default to 1000 (the same default batch size as activerecord)

### Test Plan
- The changes are covered by spec
- ci

sibaranov reviewed (approved):
Good thing to have, although it can't reduce more than 10% of current trips since 90% are finds that fetch 1 record

donaldong commented:
> Good thing to have, although it can't reduce more than 10% of current trips since 90% are finds that fetch 1 record

Good point! perhaps we can add a script to return full objects back.

donaldong commented:
rebase

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/redcord/compare/87f163e2449a91ddda79148b6060df9b50ee0598..fc488cc7c212b050f6b8d4a7f6f7acfba3cf4774).
